### ISSUE TITLE
Is there any way to wait until after files are uploaded to finish the task?

### ISSUE THREAD

**Issue by ivanmayes** [nitaigao/gulp-s3#48]
Trying to pipe some other methods based on these images being already uploaded, any way for it to wait or succeed after they finish?

**Comment by rubencaro** [comment]
+1

**Comment by davidgnin** [comment]
+1

**Comment by mkatogi** [comment]
+1

**Comment by tomdidom** [comment]
+1

**Comment by frank-weindel** [comment]
Everyone. Check out [gulp-pkgcloud-upload](https://github.com/frank-weindel/gulp-pkgcloud-upload). I wrote it to fix this very same issue that gulp-cloudfiles (the Rackspace fork of gulp-s3) had. In the process I also made it more generic: it should be able to upload to any Storage provider supported by pkgcloud. [This page](https://github.com/pkgcloud/pkgcloud/blob/master/docs/providers/amazon.md#using-storage) has an example of what the config should look like for S3. Let me know if it works for you. I've only tested it on Rackspace.
